### PR TITLE
ENH: Reworks CA download process and adds state testing

### PR DESCRIPTION
**Pull request by cc7768**
I thought it had county testing... Ergh.

### CODE REVIEW

File: src/cmdc_tools/datasets/official/CA/data.py [hunk 1]
```diff
@@ -77,5 +102,20 @@ def get_hospital_data(self):
...
+
+    def get_test_data(self):
+        resource_id = "b6648a0d-ff0a-4111-b80b-febda2ac9e09"
+        df = self.data_from_api(resource_id=resource_id)
+
+        df = df.rename(columns={"date": "dt", "tested": "tests_total",}).drop(
+            "_id", axis=1
+        )
+        df["dt"] = pd.to_datetime(df["dt"])
+        df["fips"] = self.state_fips
```
Comment (sglyon) [inline src/cmdc_tools/datasets/official/CA/data.py]: I don't think this works as a method on the county dataframe.

```
In [4]: df
Out[4]:
            county         dt  fips    value variable_name                   vintage
0      Santa Clara 2020-03-18   NaN      151   cases_total 2020-07-31 00:00:00+00:00
1      Santa Clara 2020-03-19   NaN      183   cases_total 2020-07-31 00:00:00+00:00
2      Santa Clara 2020-03-20   NaN      246   cases_total 2020-07-31 00:00:00+00:00
3      Santa Clara 2020-03-21   NaN      269   cases_total 2020-07-31 00:00:00+00:00
4      Santa Clara 2020-03-22   NaN      284   cases_total 2020-07-31 00:00:00+00:00
...            ...        ...   ...      ...           ...                       ...
61526          NaN 2020-07-25   6.0  7168139   tests_total 2020-07-31 00:00:00+00:00
61527          NaN 2020-07-26   6.0  7296578   tests_total 2020-07-31 00:00:00+00:00
61528          NaN 2020-07-27   6.0  7417866   tests_total 2020-07-31 00:00:00+00:00
61529          NaN 2020-07-28   6.0  7517466   tests_total 2020-07-31 00:00:00+00:00
61530          NaN 2020-07-29   6.0  7633840   tests_total 2020-07-31 00:00:00+00:00
```

Notice it fills the county as NaN

I think this method needs to be in its own class for CA state data

File: src/cmdc_tools/datasets/official/CA/data.py [hunk 2]
```diff
@@ -1,37 +1,57 @@
...
 
         return df
 
     def get_case_death_data(self):
-        url = self.base_url + "590188d5-8545-4c93-a9a0-e230f0db7290/resource/"
-        url += "926fd08f-cc91-4828-af38-bd45de97f8c3/download/"
-        url += "statewide_cases.csv"
-        df = pd.read_csv(url, parse_dates=["date"])
+        resource_id = "926fd08f-cc91-4828-af38-bd45de97f8c3"
+        df = self.data_from_api(resource_id=resource_id)
```
Comment (sglyon) [inline src/cmdc_tools/datasets/official/CA/data.py:86]: I'll be honest, I  don't totally understand the move to an api vs downloading the file.

Was this broken using the file download?

**Review by sglyon — CHANGES_REQUESTED** [review]

File: src/cmdc_tools/datasets/official/CA/data.py (hunk 1 above)
Comment (cc7768) [reply src/cmdc_tools/datasets/official/CA/data.py]: Good catch. Updated.

File: src/cmdc_tools/datasets/official/CA/data.py (hunk 2 above)
Comment (cc7768) [reply src/cmdc_tools/datasets/official/CA/data.py:86]: Not broken. I felt the API would likely be more stable than the csv download link, but it wasn't crucial.

**Pull request merged**